### ISSUE TITLE
Research spike: Run CI integration tests on GPU runners

### ISSUE THREAD

**Issue by deprit** [twosixlabs/armory-library#186]
Given the differences in Armory-Library dependencies installed on CPU versus GPU instances, we would like to make our CI tests more comprehensive by evaluating them on GPU-enabled runners. The research spike should answer the questions below.

1. What subset and on what actions should CI tests run on GPU instances? Should we create a separate workflow to reduce the frequency (and cost)?
2. Is it possible and how expensive would it be to use [Github Actions GPU runners](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/about-larger-runners)?
3. Should we [host our own GPU runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) on noether to reduce cost? How difficult and how much maintenance does this require?

**Comment by deprit** [comment]
As pointed out by @treubig26 and @jay-soni-26, we should reject option 3, self-hosted runners, as a [security risk for a public repository](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security).

**Comment by deprit** [comment]
From @jay-soni-26, cost estimate below for option 2, Github GPU runners.

Our builds take about 20 minutes. With the same amount of time on a Linux 4-core GPU that would be $1.40/build at $0.07/minute ([https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#per[…]tes](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates)). We currently seem to do an automated daily build and then also builds for PRs. If we did 60 builds a month, that would be $84.00. It's not clear how storage cost is calculated but we don't seem to use much so I'd assume not much. If we started out with $100, I'm pretty sure that would be more than enough for a month.
A couple optimizations to reduce cost: not doing the automated daily builds on the GPU and/or reducing frequency of the automated builds.
